rpcs3/Emu/Cell/lv2/sys_vm.cpp

```diff
@@ -135,7 +135,7 @@ error_code sys_vm_return_memory(u32 addr, u32 size)
 
 	std::lock_guard lock(block->mutex);
 
-	if (block->psize < 0x100000 + size)
```

Comment:
Using functional cast isn't good, and there is also no point in replacing < with >=.
u64{x} can be used instead, and also check the rules for integer promotion. It's only necessary to promote only one part of the expression to u64, the other part will promote implicitly.